test/unit/specs/core/createEventManager.spec.js

```diff
@@ -118,21 +119,21 @@ describe("createEventManager", () => {
     });
 
     it("does not send event when event.shouldSend returns false", () => {
-      const eventToIgnore = jasmine.createSpyObj("event", {
+      event = jasmine.createSpyObj("event", {
         finalize: () => {},
         shouldSend: false
       });
-      return eventManager.sendEvent(eventToIgnore).then(() => {
+      return eventManager.sendEvent(event).then(() => {
         expect(sendEdgeNetworkRequest).not.toHaveBeenCalled();
       });
     });
 
     it("does send event when event.shouldSend returns true", () => {
-      const eventToIgnore = jasmine.createSpyObj("event", {
+      event = jasmine.createSpyObj("event", {
         finalize: () => {},
         shouldSend: true
       });
-      return eventManager.sendEvent(eventToIgnore).then(() => {
+      return eventManager.sendEvent(event).then(() => {
```

Comment:
You'll need to add tests/assertions regarding what happens when `event.finalize` throws an error and when `event.shouldSend` returns `false`. For example, you'll need to check that the various failure callbacks get called in both cases. In the case that `event.finalize` throws an error, we'll need to make sure that the promise returned from `eventManager.sendEvent` gets rejected. In the case that `event.shouldSend` returns `false`, we'll need to make sure the promise returned from `eventManager.sendEvent` gets resolved with `undefined`.

We also need functional tests for these things. Set up a scenario where you configure Alloy with `onBeforeEventSend` throwing an error and make sure no request goes out when you attempt to send an event and that the promise you receive as a customer gets rejected. Set up another scenario where you configure Alloy with `onBeforeEventSend` returning `false` and make sure no request goes out when you attempt to send an event and that the promise you receive as a customer gets resolved with an object (even though you resolve the promise being returned from eventManager with `undefined`, there's code higher up in Alloy that ensures that all promises returned from commands are resolved with an object, even if it's an empty object).